src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -253,9 +253,13 @@ private ResultSetWrapper getNextResultSet(Statement stmt) throws SQLException {
     try {
       if (stmt.getConnection().getMetaData().supportsMultipleResultSets()) {
         // Crazy Standard JDBC way of determining if there are more results
-        if (!((!stmt.getMoreResults()) && (stmt.getUpdateCount() == -1))) {
+        if (stmt.getMoreResults() || stmt.getUpdateCount() != -1) {
```

Comment:
@happysearch 

This change may not be right.
See the javadoc of [java.sql.Statement#getMoreResults()](https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html#getMoreResults--).
Here is an excerpt:

> There are no more results when the following is true:
>     // stmt is a Statement object
>     ((stmt.getMoreResults() == false) && (stmt.getUpdateCount() == -1))

It means, I believe, we need to check the results of both methods to make the decision.
If you revert this part, does it still fix your problem?